.gitlab-ci.yml

```diff
@@ -1,14 +1,116 @@
-test:
-  image: "python:3.6-stretch"
-  script:
-    - apt-get update -qq
-    - apt-get install -y git
-    - apt-get install -y xsltproc
-    - apt-get install -y genisoimage
-    - apt-get install -y enchant
-    - apt-get install -y shellcheck
-    - apt-get install -y python3-pip
-    - pip3 install --upgrade pip
-    - pip install tox
-    - tox -e check
-    - tox -e unit_py3_6 "-n $(nproc)"
+stages:
+  - test
+  - doc
+  - package
+
+.fedora_install_deps: &fedora_install_deps
+  before_script:
+    - dnf install -y --refresh python3 python3-devel 'python3dist(pip)'
+    # requirements
+    - dnf install -y 'python3dist(docopt)' 'python3dist(lxml)' 'python3dist(pyyaml)' 'python3dist(pyxattr)' 'python3dist(six)' 'python3dist(future)' 'python3dist(requests)'
+    # dev-requirements
+    - dnf install -y 'python3dist(pytest)' 'python3dist(pytest-xdist)' 'python3dist(pytest-cov)' 'python3dist(mock)' 'python3dist(flake8)' 'python3dist(sphinx)' python3-sphinx_rtd_theme 'python3dist(sphinxcontrib-spelling)' 'python3dist(pyenchant)' 'python3dist(tox)' make gcc which xz xorriso
+
+.distro_test: &distro_test
+  script:
+    - python3 setup.py build
+    - export PYTHONPATH=$(pwd)
+    - cd test/unit/ && pytest-3.7 --no-cov-on-fail --cov=kiwi --cov-report=term-missing --cov-fail-under=100 --cov-config .coveragerc -n $(nproc)
```

Comment:
I think I don't understand why we have the code from tox.ini that calls pytest running copied and called in extra test here ? Can you explain the motivation ?

Reply:
It's left over from my initial experimentation with gitlab-ci. And IIRC I had some problems with pytest on Fedora where `pytest` doesn't exist, but `pytest-3.7` does.

But yes, this is something that I wanted to remove (and forgot about it).

Reply:
Fedora doesn't _generally_ use alternatives to manage the unversioned names of the binaries.